test/test.c

```diff
@@ -24242,6 +24242,38 @@ test_JetStreamMgtConsumers(void)
         cfg.FilterSubject = "bar";
     }
 
+#define TIME_20350101 (2051251200L * 1000000000L)
+#define TIME_20350101_STR "\"2035-01-01T08:00:00Z\""
+
+    if (serverVersionAtLeast(2, 11, 0))
+    {
+        test("Add consumer (non durable, paused): ");
+        cfg.PauseUntil = TIME_20350101;
+        s = js_AddConsumer(&ci, js, "MY_STREAM", &cfg, NULL, &jerr);
+        testCond((s == NATS_ERR) && (jerr == JSStreamNotFoundErr) && (ci == NULL));
+        nats_clearLastError();
+
+        test("Verify config: ");
+        s = natsSubscription_NextMsg(&resp, sub, 1000);
+        testCond((s == NATS_OK) && (resp != NULL) && (strncmp(natsMsg_GetData(resp), "{\"stream_name\":\"MY_STREAM\","
+                                                                                     "\"config\":{\"deliver_policy\":\"last\","
+                                                                                     "\"description\":\"MyDescription\","
+                                                                                     "\"deliver_subject\":\"foo\","
+                                                                                     "\"opt_start_seq\":100,"
+                                                                                     "\"opt_start_time\":\"2021-06-23T18:22:00.12345Z\",\"ack_policy\":\"explicit\","
+                                                                                     "\"ack_wait\":200,\"max_deliver\":300,\"filter_subject\":\"bar\","
+                                                                                     "\"metadata\":{\"key1\":\"val1\",\"key2\":\"val2\"},"
+                                                                                     "\"pause_until\":" TIME_20350101_STR ","
+                                                                                     "\"replay_policy\":\"instant\",\"rate_limit_bps\":400,"
+                                                                                     "\"sample_freq\":\"60%%\",\"max_waiting\":500,\"max_ack_pending\":600,"
+                                                                                     "\"flow_control\":true,\"idle_heartbeat\":700,"
+                                                                                     "\"num_replicas\":1,\"mem_storage\":true}}",
+                                                              natsMsg_GetDataLength(resp)) == 0));
+        natsMsg_Destroy(resp);
+        cfg.PauseUntil = 0;
+        resp = NULL;
+    }
```

Comment:
I think you may also want to have a test that verifies that a consumer CAN be created paused. I think that it is possible.